[GENESIS, PVM] Remove genesis.MultisigAlias, move corresponding methods and tests

evlekht opened the pull request:
## Why this should be merged
Currently, we have 3 places with msig alias definitions:

1. `vms/platformvm/genesis/camino.go` (used in genesis config parsing, used in platformVM genesis)
```
type MultisigAlias struct {
    Alias     ids.ShortID   `serialize:"true" json:"alias"`
    Threshold uint32        `serialize:"true" json:"threshold"`
    Addresses []ids.ShortID `serialize:"true" json:"addresses"`
    Memo      string        `serialize:"true" json:"memo"`
}
```
2. `vms/components/multisig/camino_multisig.go (used in state)`
```
type Alias struct {
    ID     ids.ShortID
    Memo   types.JSONByteSlice `serialize:"true" json:"memo"`
    Owners verify.Verifiable   `serialize:"true" json:"owners"`
}
```
3. Plain fields in tx `vms/platformvm/txs/camino_multisig_alias_tx.go`
```
type MultisigAliasTx struct {
    BaseTx `serialize:"true"`
    Alias ids.ShortID `serialize:"true" json:"alias"`
    Owner fx.Owner `serialize:"true" json:"owner"`
    ChangeAuth verify.Verifiable `serialize:"true" json:"changeAuthorization"`
}
```
All msig verify logic only exists for the `vms/platformvm/genesis/camino.go` and will always use actual msig fields for id computation which then is used to verify actual id.
But we'll also need to verify msig definition in (3) when we'll add featur to add new msig aliases via tx.
And we also want msig alias id to be independent from msig alias owner addresses, because we'll add feature to modify existing aliases without changing alias id itself.
Thats a lot of inconsistency and bad logic placement.
## How this works
Remove genesis.MultisigAlias in order to keep logic in one type, move verify method to multisig.Alias type and make it use owners own verify method.
Create parsed config msig alias type, move computeID to genesis config MultisigAlias type, cause its used only there.
Move tests to corresponding packages. Tests that were testing secp256k1fx.OutputOwners inside msig alias were removed, cause now msig alias verify calls corresponding secp256k1fx.OutputOwners.Verify method which has its own tests.
Create state type in order to not store alias id as part of the value, cause its stored as key. That is needed because multisig.Alias now serializes id field as it will be part of tx.
## How this was tested
Unit-tests, manual tests with one local node (breakpoints + api calls).

charalarg commented on vms/components/multisig/camino_multisig.go:
```diff
@@ -4,13 +4,42 @@
 )
 
+// we can't use avax.MaxMemoSize because for some reason it results in import cycle even if it isn't
+
```
I think the import cycle exists:
-> multisig imports avax (If we add avax.MaxMemoSize)
-> avax imports secp256k1fx
-> secp256k1fx imports multisig in camino_fx

but IMO for the time being we should just remove this comment instead of trying to resolve the import cycle just for this const.

evlekht replied:
You are right, this is the reason (I checked)
And its actually bad - only avax (and one of ours) test files import secp256k1fx - its not intended by implementation. That ruins package usage a bit.

charalarg replied:
Yep, true.. But still avax code that we shouldn't touch :) So we can just redefine the memoSize as you did.

evlekht committed:
[GENESIS, PVM] Remove genesis.MultisigAlias in order to keep logic in one type

Move computeID and verify methods to corresponding types, move tests, create parsed config msig alias type, create state type (in order to not store alias id as part of the value, cause its stored as key)
